src/components/form/super_select/super_select_control.js

```diff
@@ -33,7 +33,7 @@ export const EuiSuperSelectControl = ({
   // React HTML input can not have both value and defaultValue properties.
   // https://reactjs.org/docs/uncontrolled-components.html#default-values
   let selectDefaultValue;
-  if (!value) {
+  if (!value && value !== '') {
```

Comment:
Instead, let's change that original `!value` to `value == null` to avoid JavaScript's truthiness weirdness. This solves this `''` use case but also any other, as value can be any type of React node  like `false` or `0`

```suggestion
  if (value == null) {
```